manual/manual/refman/exten.etex

```diff
@@ -621,92 +619,98 @@ Each implementation is a structure that we can encapsulate as a
 first-class module, then store in a data structure such as a hash
 table:
 \begin{caml_example*}{verbatim}
-        module type DEVICE = sig [@@@ellipsis] end
-        let devices : (string, (module DEVICE)) Hashtbl.t = Hashtbl.create 17
-
-        module SVG = struct [@@@ellipsis] end
-        let _ = Hashtbl.add devices "SVG" (module SVG : DEVICE)
-
-        module PDF = struct [@@@ellipsis] end
-        let _ = Hashtbl.add devices "PDF" (module PDF: DEVICE)
+[@@@ellipsis.start]
+type picture = unit
+let parse_cmdline () : string = assert false
+[@@@ellipsis.stop]
+module type DEVICE = sig [@@@ellipsis.start]val draw : picture -> unit[@@@ellipsis.stop] end
+let devices : (string, (module DEVICE)) Hashtbl.t = Hashtbl.create 17
+
+module SVG = struct let draw () = ()[@@ellipsis] end
+let _ = Hashtbl.add devices "SVG" (module SVG : DEVICE)
+
+module PDF = struct let draw () = ()[@@ellipsis] end
+let _ = Hashtbl.add devices "PDF" (module PDF: DEVICE)
```

Comment:
@Octachron I think that you may find this block of code interesting.

The first ellipsis is used in place of a separate `{caml_eval}` block as I had in my previous version (before the last rebase), and it suggests that maybe longer-term we could think of using `[@hide]` attributes that are completely removed instead of a separate environment.

The second ellipsis has this form, instead of a signature-item attribute, because of the issue I reported in https://github.com/ocaml/ocaml/pull/1765/files#r186431706. I am trying to come with a fix for the issue, but maybe we could consider trying to get this PR merged before that -- it's been long enough in the baking.

Reply:
I agree that it would be nice to merge soon. I will do a full review by tomorrow.

> that maybe longer-term we could think of using [@hide] attributes that are completely removed instead of a separate environment.

This is definitively doable. At the same time, I fear that mixing hidden and non-hidden code fragment too freely might be confusing for the reader.

Reply:
Thinking more about it, I think we should wait until we agree on what to do with #1765 before merging.